drools-parser/src/main/java/org/drools/parser/ParserStringUtils.java

```diff
@@ -38,4 +38,8 @@ public static String getTokenTextPreservingWhitespace(ParserRuleContext ctx, Tok
         // tokenStream is required to get hidden channel token (e.g. whitespace). Unlike getTextPreservingWhitespace, this method reflects Lexer normalizeString
         return tokenStream.getText(ctx.start, ctx.stop);
     }
+
+    public static String trimThen(String rhs) {
+        return rhs.substring("then".length());
```

Comment:
Hi @tkobayas: my understanding is that this method is specially crafted to remove leading `then` from a string, which could be a good idea; but there is not any check/verification: its correct execution depends completely on calling code.
I would add some check to verify that the string actually starts with `then`: wdyt ?

Reply:
@gitgabrio Thank you for the suggestion! Added the check.